runtime/str.go

```diff
@@ -453,6 +453,37 @@ func strNew(f *Frame, t *Type, args Args, _ KWArgs) (*Object, *BaseException) {
 	return s.ToObject(), nil
 }
 
+// strReplace returns a copy of the string s with the first n non-overlapping
+// instances of old replaced by new. If old is empty, it matches at the
+// beginning of the string. If n < 0, there is no limit on the number of
+// replacements.
+func strReplace(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	// TODO: Support unicode replace.
+	expectedTypes := []*Type{StrType, StrType, StrType, ObjectType}
+	argc, n := len(args), -1
+	if argc == 3 {
+		expectedTypes = expectedTypes[:argc]
+	}
+	if raised := checkMethodArgs(f, "replace", args, expectedTypes...); raised != nil {
+		return nil, raised
+	}
+	if argc == 4 {
+		o, raised := IntType.Call(f, Args{args[3]}, nil)
+		if raised != nil {
+			return nil, raised
+		}
+		n = toIntUnsafe(o).Value()
+	}
+	s := toStrUnsafe(args[0]).Value()
+	old := toStrUnsafe(args[1]).Value()
+	new := toStrUnsafe(args[2]).Value()
+	// CPython only, pypy supposed to be same as Go
+	if len(s) == 0 && len(old) == 0 && n >= 0 {
+		return NewStr("").ToObject(), nil
+	}
+	return NewStr(strings.Replace(s, old, new, n)).ToObject(), nil
```

Comment:
I think strings.Replace() may not be compatible with the Python version because when `old == ""`, strings.Replace: `matches at the beginning of the string and after each UTF-8 sequence` whereas in Python it will match after each *byte*. So if the string contains UTF-8 sequences, the results will differ (this is a good test case to have, as well).